os/drivers/testcase/test_sched.c

```diff
@@ -0,0 +1,146 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <errno.h>
+#include <debug.h>
+#include <tinyara/testcase_drv.h>
+#include <tinyara/sched.h>
+#include "kernel_test_proto.h"
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+struct tcb_s *tcb;
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+int test_get_self_pid(int cmd, unsigned long arg)
+{
+	tcb = sched_self();
+	if (tcb == NULL) {
+		return ERROR;
+	}
+	return tcb->pid;
+}
+
+int test_is_alive_thread(int cmd, unsigned long arg)
+{
+	int ret;
+	ret = get_tcb(arg);
+	if (ret == ERROR) {
+		return ERROR;
+	}
+	return OK;
+}
+
+int test_get_tcb_sigprocmask(int cmd, unsigned long arg)
+{
+	int ret;
+	ret = get_tcb(arg);
+	if (ret == ERROR) {
+		return ERROR;
+	}
+	return tcb->sigprocmask;
+}
+
+int test_get_tcb_adj_stack_size(int cmd, unsigned long arg)
+{
+	int ret;
+	ret = get_tcb(arg);
+	if (ret == ERROR) {
+		return ERROR;
+	}
+	return tcb->adj_stack_size;
+}
+
+#ifdef CONFIG_TC_KERNEL_ROUNDROBIN
+int test_get_tcb_timeslice(int cmd, unsigned long arg)
+{
+	int ret;
+	ret = get_tcb(arg);
+	if (ret == ERROR) {
+		return ERROR;
+	}
+	return tcb->timeslice;
+}
+#endif
+
+int test_sched_foreach(int cmd, unsigned long arg)
+{
+	sched_foreach((void *)arg, NULL);
+}
+
+int get_tcb(unsigned long arg)
```

Comment:
Is this function really needed?

Reply:
actually it is not necessary.
I will remove this function.